packages/components/src/popover/index.js

```diff
@@ -273,14 +274,29 @@ const Popover = ( {
 				contentRect.current = contentRef.current.getBoundingClientRect();
 			}
 
+			const { offsetParent } = containerRef.current;
+			let relativeOffsetTop = 0;
+
+			if ( offsetParent ) {
+				const offsetParentRect = offsetParent.getBoundingClientRect();
+
+				relativeOffsetTop = offsetParentRect.top;
+				anchor = new window.DOMRect(
+					anchor.left - offsetParentRect.left,
+					anchor.top - offsetParentRect.top,
+					anchor.width,
+					anchor.height
+				);
+			}
```

Comment:
I believe this assumes the parent is relatively positioned. Should it be opt-in, won't this break popovers if it's not the case?

Reply:
There's a check for `offsetParent`. That should be enough?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent

Interesting that `offsetParent` does seem to return the body element in Firefox. I'll adjust the check.

Reply:
Does the logic here depend that the developer apply a style to their popover to position it absolutely? Is that what you're saying by checking if the position is absolute? Otherwise, it doesn't seem very self-contained, or clear why this logic would exist here if it's something only relevant for custom popover behavior. Can we apply `position: absolute` within this component when it's appropriate? Is there any opportunity to be be smart about detecting that the popover is being rendered in a scrollable container?

Reply:
Sure, we could also introduce a (perhaps unstable) prop to change the positioning. This logic has little to do with the scrollable container, but more with the fact that the popover has to be  positioned relatively within it. You could do the same thing outside a nested scroll container if you'd want the popover to scroll. All popovers are currently blocking scroll.

To reproduce: open a paragraph inspector, and open the color settings panel. Hover over a color and then move over the tooltip. Scrolling is now blocked.

Reply:
![popover-scroll](https://user-images.githubusercontent.com/4710635/72810413-cac8be00-3c5d-11ea-9fb2-3e2688127057.gif)

Reply:
Hm, okay, I might not be following, but I'd wondered if this is something where we should change the default behavior more to like what you're proposing here currently as being specific to the sibling inserter. In other words, can all popovers be absolutely positioned within their closest positioned ancestor? Or is this a problem for popovers which have their own scroll (modals)?

Reply:
I think the behaviour is expected in some popovers, so I don't think we can change change it for all. E.g. it does seem useful for the popovers of buttons inside the toolbar.

How about a prop to disable the behaviour. I'm not sure what to call it though.

Reply:
Are you okay with the current state of the PR for now? It would be good to eventually add a proper option etc. for sure

Reply:
I'm okay on the point of introducing more formal support later. It's not entirely clear to me if I would expect some impact of these changes on existing usage which doesn't use the absolute-positioned approach. You mentioned earlier "There's a check for `offsetParent`", but I don't understand specifically how that's meant to help.

In any case, it seems from some brief testing that things work well enough.

Reply:
I guess it assumes that elsewhere, the Slot is rendered such that there wouldn't be an `offsetParent` ? Technically we also support an "in-place rendering" option, when there is no Slot context available. I don't think we use it anywhere ourselves. But it seems it could be impacted by these changes.

Reply:
In all other cases, the popover has fixed positioning, even when rendered in place, which means that it is offset relative to the viewport and that it doesn't have an offset parent. When the popover has absolute positioning, the offset parent is the first ancestor that has a relative position.

Reply:
Gotcha. I think I misunderstood how `offsetParent` was being used here. I see now it's used as an indicator of the component itself being fixed, so only if someone went out of their way to manually override the fixed property would it ever have an impact 👍

Reply:
I guess the comment could be clearer. 😄